ipatests/test_integration/test_trust.py

```diff
@@ -992,3 +992,270 @@ def test_server_option_with_unreachable_ad(self):
             self.master.run_command(['rm', '-f', ad_zone_file])
             tasks.configure_dns_for_trust(self.master, self.ad)
             self.remove_trust(self.ad)
+
+
+class TestNonPosixADTrustWithAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+
+    def test_add_nonposix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust'])
+
+    def add_idrange_auto_private_group(
+        self, rangename, trust_type, auto_priv_grp_val
+    ):
+        """
+        Function to add additional idrange
+        """
+        add_range = 100000
+        range_name = self.ad_domain.upper() + "_id_range"
+        domainname = self.ad_domain.upper()
+        range_dn = DN(
+            ("cn", range_name),
+            ("cn", "ranges"),
+            ("cn", "etc"),
+            self.master.domain.basedn,
+        )
+        conn = self.master.ldap_connect()
+        entry = conn.get_entry(range_dn)  # pylint: disable=no-member
+        baseid = entry.single_value.get("ipaBaseID")
+        range = entry.single_value.get("ipaIDRangeSize")
+        cal_baseid = int(baseid) + int(range) + add_range
+        newbaseid = str(cal_baseid)
+        cmd = [
+            "ipa",
+            "idrange-add",
+            rangename,
+            "--type",
+            trust_type,
+            "--base-id",
+            newbaseid,
+            "--range-size",
+            range,
+            "--rid-base",
+            newbaseid,
+            "--dom-name",
+            domainname,
+        ]
+
+        if trust_type == "ipa-adtrust":
+            cmd.extend(("--type", "ipa-ad-trust"))
+        elif trust_type == "ipa-ad-trust-posix":
+            cmd.extend(("--type", "ipa-ad-trust-posix"))
+
+        if auto_priv_grp_val == "false":
```

Comment:
Remove if elif else conditions.
Instead of that directly add  
`cmd.extend(("--auto-private-groups", auto_priv_grp_val))`
Also define default value for `auto_priv_grp_val`